lib/OpenQA/Scheduler/Scheduler.pm

```diff
@@ -151,6 +150,11 @@ sub job_grab {
     my $workercaps = $args{workercaps};
 
     my $worker = _validate_workerid($workerid);
+    if ($worker->job) {
+        # TROUBLE ahead - do not call seen, to get into dead worker detection first
```

Comment:
if we update t_updated on every grab, the worker will never get dead - and we won't restart the job